### ISSUE TITLE
Pickup points not always saved on checkout

### ISSUE THREAD

**Issue by joosev** [Pakettikauppa/woo-pakettikauppa#181]
Sometimes people can complete checkout process without selected pickup point saving.

I have not been able to replicate but the problem is occurring on multiple websites and even on those websites I can't replicate the problem so it seems to be related to something user does.

**Comment by joosev** [comment]
My initial though is that order is saved by the payment module and that makes the selected pickup point disappear. So selected pickup point should be saved somehow to somewhere so it can be fetched later.

**Comment by joosev** [comment]
For instance payment module sends server->server request "this order is paid" and has it's custom create order implemented. And if selected pickuip point is saved in the browser, that data is not available.

**Comment by joosev** [comment]
One time this happened:
User tried to make an order -> it failed for reason unknown.

Then user tried to make order again and everything was prefilled (not known was it browser prefilled or woocommerce prefilled) and no pickup point chooser was shown. Although Woocommerce let user pass the checkout without choosing the pickup point.

**Comment by joosev** [comment]
Some of the orders that were cancelled first time, was really cancelled for example in the payment phase. The reason of cancelling of orders was a UX problem in the ecommerce site where clients did not know where to input discount code. So first order was cancelled, then next order had information already prefilled but no pickup point chooser was visible.

**Comment by mainiomooses** [comment]
[Video of issue replicated w/ Firefox](https://youtu.be/z_rfCXU_QKw)

I was able to replicate this some time back using Firefox. After populating all the checkout fields or while populating the checkout fields Woo AJAX would not fire (update_checkout). Thus checkout would not get updated with info that feeds into the calculated shipping part of checkout. 

If that happens, there's no need to configure shipping in any way and where Woo's concerned, everything is fine as every field required has been filled.

This is also what I've heard from my customer's customers.

This was also the case in a collegue's customer's store - the store owned contacted the customer whose pickup point info was not present BUT whose shipping address details (on which Woo AJAX reacts to) were correctly input - indeed the customer never saw a field for selecting a pickup point nor was he/she prompted one.

TLDR; Woo AJAX does not correctly fire in some use cases of the checkout process / form with some browsers; at least on Firefox.

**Comment by joosev** [comment]
This issue still exists with 2.4.0 version

**Comment by joosev** [comment]
There is a possibility, that nonce checks fails for some reason for some merchants and letting users pass validation that way.

**Comment by joosev** [comment]
This is maybe solved for now.
